project-koku/koku
# COST-2670  Org Units showing correct total cost for each sub org

## Jira Ticket

[COST-2670](https://issues.redhat.com/browse/COST-2670)

## Description

This change will ...

add check on type and value for request.user.admin 
add function to get sub organization units hierarchy, to fix access list

## Testing

1. Checkout Branch
2. Restart Koku
3. Hit the following endpoints in separate windows
   
    `127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?delta=cost&filter[limit]=10&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[org_unit_id]=OU_001&order_by[cost]=desc&cost_type=unblended_cost`

5. Update DEFAULT_IDENTIY in settings with the values below and repeat steps 3 to 4 
```
"is_org_admin": "False", 
"access": {
        "aws.account": {"read": ["*"]}, 
        "aws.organizational_unit": {"read": ["OU_001"]}
   }
```

6. You should see consistent sub org unit cost with for a user with admin access or a user with restricted access privileges.

## Notes
since `OU_005` is a sub org of `OU_001` you can hit the endpoint below to verify the sub-org unit cost total

`127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/?delta=cost&filter[limit]=10&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[org_unit_id]=OU_005&order_by[cost]=desc&cost_type=unblended_cost`

...
